### PR TITLE
Kotlin Cleanup in TemplateFilters.kt

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
refactoring code in TemplateFilters.kt

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt [hunk 1]
```diff
@@ -62,15 +62,14 @@ object TemplateFilters {
      * @return Result of filter on current txt.
      */
     @KotlinCleanup("maybe change var to val")
-    fun apply_filter(txtInput: String, filterInput: String, field_name: String, tag: String): String {
+    fun applyFilter(txtInput: String, filterInput: String, field_name: String, tag: String): String {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:65]: As this class in from libanki, I think it's set to follow the upstream naming so this change should be reverted.

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt [hunk 2]
```diff
@@ -123,36 +122,35 @@ object TemplateFilters {
...
-        val m = Pattern.compile(String.format(Locale.US, CLOZE_REG, ord)).matcher(txt)
-        val repl = StringBuffer()
-        while (m.find()) {
-            // replace chosen cloze with type
-            @KotlinCleanup("maybe make non-null")
-            var buf: String?
-            buf = if (type == 'q') {
-                if (!TextUtils.isEmpty(m.group(4))) {
-                    "[" + m.group(4) + "]"
+        return if (!Pattern.compile(String.format(Locale.US, CLOZE_REG, ord)).matcher(txtInput)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:126]: This change should be reverted, there's no reason to add an extra indentation level for so much code just to have return if-else.

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt (hunk 1 above)
Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:65]: Last time I brought this up there was no clear consensus - might benefit from a vote from the other regular devs so a consistent approach can be taken moving forward.  https://github.com/ankidroid/Anki-Android/issues/11582

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:65]: feels like in libanki in particular, which really is (was?) a Python code mirror there is some merit to matching the python - it is a port, and I believe it's relegated to "running just as long as things aren't all in new schema" ?

feels like in the issue you linked (11582) there is merit in idiomatic casing, I'd characterize that as wrappers or codegen code and they are not ports, they are living code that has a future

In ported code there is some value to matching the other language and no value really in a larger diff to change it, in wrappers + codegen there's value to idiomatic casing that feels correct in the wrapping + codegenerated language

So to me this feels like diff for diffs sake with slight negative value

Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:65]: Well, my thinking here is consistency causes less surprises - a rule where camelCase is used for standard Android code and backend methods, but not manually ported methods is potentially more confusing than consistent case throughout the codebase. The desktop library code is all snake_case, so a user comparing the desktop code would need to convert the case in their heads when looking at backend methods, but not manual implementations.

Anyway, just my 2c :-)

**Pull request closed without merging**